Add Behringer-CMD-PL-1 controller

yledoare opened the pull request:
Rebased from 2.2

yledoare committed:
New Behringer-CMD-PL-1 controller

yledoare committed:
Fixes

yledoare committed:
Add Behringer-CMD-PL-1 controller

Holzhaus commented:
Thank you, pre-commit issues have been fixed. Note that you could just have force-pushed to the same branch instead of opening a new PR, but that doesn't matter now.

Starting with Mixxx 2.3, all controller mappings are now supposed to be documented in the Manual. This is necessary so that other users know how to use this mapping. We also need it to review it properly and check if the mapping makes sense and if the code matches the documentation.

Please fork the [mixxxdj/manual](https://github.com/mixxxdj/manual) repository, check out the `manual-2.3.x` branch and add documentation to the [`source/hardware/controllers` directory](https://github.com/mixxxdj/manual/tree/manual-2.3.x/source/hardware/controllers). Open the pull request and add the link to the PR description.

Here's an example how such documentation could look like: https://manual.mixxx.org/2.3/en/hardware/controllers/roland_dj_505.html (If you don't have schematic images and don't want to draw them yourself, just use the same numbering as in the controller's manual and document the buttons like that).

Holzhaus commented on res/controllers/Behringer CMD PL-1.midi.xml:
```diff
@@ -0,0 +1,75 @@
+<?xml version='1.0' encoding='utf-8'?>
+<MixxxControllerPreset mixxxVersion="2.4.0+" schemaVersion="1">
```
```suggestion
<MixxxControllerPreset mixxxVersion="2.2.0+" schemaVersion="1">
```

Holzhaus commented:
I'll convert this to draft until we have a pending manual PR.

uklotzde commented:
Please rebase on 2.3 and then force push.

I will change the milestone to 2.3.1 because I don't expect that this PR will be ready for 2.3.0.

Holzhaus committed:
skins/QMLDemo: Add cue popup

Holzhaus committed:
skins/QMLDemo: Only draw left overview channel

Holzhaus committed:
QmlWaveformOverview: Add basic support for multiple renderer styles

Holzhaus committed:
QML: Add filtered renderer to waveform overview

Holzhaus committed:
skins/QMLSkin: Add use filtered waveform overview renderer

uklotzde committed:
Synchronize file modified time with track source on metadata import/export

ronso0 committed:
Skins: fix remaining font leaks

= spots where the system font or that set by Qt Settings (qt5ct)
could screw up the skin designs (cropped labels or table header)

Holzhaus committed:
QmlPlayerProxy: Add method to load track by location to QmlPlayerProxy

Holzhaus committed:
QmlPlayerProxy: Add support for loading file:// URLs

Holzhaus committed:
skins/QMLDemo: Add support for dropping track URLs on decks

ywwg committed:
Merge branch 'main' into stopped-explicit

daschuer committed:
Merge pull request #3954 from ronso0/waveform-textmarker-font

Waveform: always use pixel size and 'Open Sans' for text markers

daschuer committed:
Make sure getSettingsPath() returns the correct directory

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

uklotzde committed:
Merge branch 'main' of git@github.com:mixxxdj/mixxx.git into dlgtrackinfo

Holzhaus committed:
QML: Add MathUtils js library

Holzhaus committed:
skins/QMLDemo: Use MathUtils library

Holzhaus committed:
QML: Fix wrong Mixxx.Controls qmldir file

Holzhaus committed:
skins/QMLDemo: Simplify slider handle x expression

Holzhaus committed:
skins/QMLSkin: Make ControlKnob's group/key an alias property

Holzhaus committed:
QML: Unify ControlKnob and ControlSlider implementations

Holzhaus committed:
QML: Set Knob value from wheel/drag event, restore binding afterwards

Holzhaus committed:
skins/QMLDemo: Add support for scaled knobs

Holzhaus committed:
QML: Use parent for Knob shape instead of explicit root name

Holzhaus committed:
QML: Add TODO comments for Mixxx.Controls

Holzhaus committed:
QmlWaveformOverview: Use simpler disconnect call

Holzhaus committed:
QmlWaveformOverview: Use QPointer for m_pPlayer member

Holzhaus committed:
QML: Add wheelStepSize property to knob

Holzhaus committed:
skins/QMLDemo: Don't enable mipmapping on Knobs for performance reasons

Holzhaus committed:
skins/QMLDemo: Improve knob styling

uklotzde committed:
Merge pull request #3919 from Holzhaus/qml-waveforms

QML: Add support for Waveform Overviews

Holzhaus committed:
skins/QMLDemo: Add support for horizontal Sliders and improve shadows

Swiftb0y committed:
QMLDemo: make MathUtils an ES6 Module

Holzhaus committed:
skins/QMLDemo: Move knob background color to theme singleton

Holzhaus committed:
skins/QMLDemo: Fix EqKnob status button

Holzhaus committed:
QML: Remove unnecessary knob grabPermissions

ronso0 committed:
Skin qss: for library widgets use class name only

ronso0 committed:
Deere: restore normal font weight for menus and some widgets (revert bold)

Holzhaus committed:
QML: Add reset() method to reset a QmlControlProxy to default value

Holzhaus committed:
skins/QMLDemo: Replace resettable slider/knob components with reset()

Swiftb0y committed:
Merge pull request #3955 from Holzhaus/qml-control-refactor

QML: Refactor Knob/Slider controls

Holzhaus committed:
QML: Add QmlConfigProxy singleton for accessing color palettes

Holzhaus committed:
skins/QMLDemo: Use actual hotcue color palette in cue popup

uklotzde committed:
Merge pull request #3956 from daschuer/migrateOldSettings

Make sure args.getSettingsPath() returns the correct directory

uklotzde committed:
Merge pull request #3957 from Holzhaus/qml-controlproxy-reset

QML: Add QmlControlProxy::reset() method

daschuer committed:
Added some debian debendencies for the new QML skin

daschuer committed:
Merge remote-tracking branch 'upstream/2.3' into main

daschuer committed:
Silence spam about not supported help-all

daschuer committed:
Merge pull request #3958 from ronso0/qssfontleaks

Skins: fix remaining 'font leaks'

uklotzde committed:
Merge pull request #3962 from daschuer/parser_warning

Silence spam about not supported help-all

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

uklotzde committed:
Merge branch 'main' of git@github.com:mixxxdj/mixxx.git into dlgtrackinfo

Holzhaus committed:
QML: Do not alias associated anchor property (for Qt 5.12 compatiblity)

ronso0 committed:
Skin qss: for library widgets use class name only pt II

Holzhaus committed:
QML: Move HotcueButton color logic to ControlProxy

ronso0 committed:
WLibrarySidebar: scale icons with library font

Holzhaus committed:
Merge pull request #3961 from daschuer/qml_debian

Added some debian dependencies for the new QML skin

ywwg committed:
Make sure the first beat in a const beat grid is as close to the start of the track as possible.

ywwg committed:
Use the beatlength of the rounded bpm, not the center.

daschuer committed:
Added an help text to debian and macos_buildenv

Holzhaus committed:
skins/QMLDemo: Move Hotcue and HotcuePopup to separate components

Holzhaus committed:
res/skins/QMLDemo:  Improve HotcuePopup close policy

Co-authored-by: Swiftb0y <12380386+Swiftb0y@users.noreply.github.com>

Holzhaus committed:
pre-commit: Replace required properties automatically

Holzhaus committed:
QML: Remove required property for Qt 5.12 compat

ronso0 committed:
Skins: move common properties to default.qss

* menubar checkboxes
* lib BPM lock
* Played checkbox
* menu arrows
* menu separators

ronso0 committed:
Shade (fix): use Open Sans in table header

Holzhaus committed:
res/skins/QMLDemo: Add missing HotcueButton activate binding

ronso0 committed:
Skin context menus: scale checkboxes with font size

ronso0 committed:
Skins, context menus: move common properties to default.qss

Holzhaus committed:
pre-commit: Use main.cpp for getting clang_format config

That file name is less likely to change. Previously, `mixxx.cpp`
was used, which might be renamed in a later refactoring.

Holzhaus committed:
tools/githelper: Fix wrong docstring

Holzhaus committed:
tools/githelper: Fix misdetection of changed lines when file was moved

Holzhaus committed:
tools/githelper: Actually use the filtered list of included files

ronso0 committed:
Skins: use SVGs for sidebar branch icons

ronso0 committed:
Skins: fix menubar checkbox selection highlight on Linux

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

uklotzde committed:
Merge branch 'main' of git@github.com:mixxxdj/mixxx.git into dlgtrackinfo

uklotzde committed:
DlgTrackInfo: Fix data binding for key text field

ywwg committed:
Add comment

uklotzde committed:
Linux: Remove obsolete X11 code from Qt 4

This should have been accomplished by using Qt::AA_X11InitThreads,
which became obsolete with Qt5.

Holzhaus committed:
skins/QMLDemo: Use async coverart loading

ronso0 committed:
Skins, context menus: dim all disabled menu items

Swiftb0y committed:
Merge pull request #3952 from Holzhaus/qml-deck-drop

QML: Add support for dropping tracks on Decks

daschuer committed:
Merge pull request #3963 from Holzhaus/qml-assoc-property-alias-fix

QML: Do not alias associated anchor property (for Qt 5.12 compatiblity)

Holzhaus committed:
QML: Use x axis in knob mouse interaction code like legacy skins do

daschuer committed:
remove redundant shift call

ywwg committed:
fix first beat: Use fmod

daschuer committed:
Merge pull request #3965 from ywwg/fix-first-beat

Fix const beat grid first beat placement

daschuer committed:
Advance beat analysis version to V4

... because of the moved first beat in const beat grits introduced in #3965

daschuer committed:
Merge remote-tracking branch 'upstream/main' into qmlformat

Holzhaus committed:
Merge pull request #3906 from uklotzde/dlgtrackinfo

DlgTrackInfo: Use TrackRecord as model

uklotzde committed:
Merge pull request #3971 from Holzhaus/qml-knob-mouse-interaction

QML: Use x axis in knob mouse interaction code like legacy skins do

Swiftb0y committed:
Merge pull request #3959 from Holzhaus/qml-cuepopup

QML: Add support for reading color palettes from config and add cue popup

uklotzde committed:
Merge pull request #3973 from daschuer/roundingVersionV4

Advance beat analysis version to V4

Swiftb0y committed:
Merge pull request #3972 from Holzhaus/coverart-async

skins/QMLDemo: Use async coverart loading

Holzhaus committed:
test: Only register SoundSourceProviders when actually necessary

This allows tests to inherit from MixxxTest and still register
SoundSourceProviders themselves, e.g. by instantiating CoreServices.
It also reduces log spam caused by the sound source provider
registration.

Holzhaus committed:
Merge pull request #3970 from uklotzde/obsolete-x11initthreads

Linux: Remove obsolete X11 code from Qt 4

daschuer committed:
Add a comment spelling the Qt version literaly

Swiftb0y committed:
update 2.3 Changelog

ronso0 committed:
mention cloning from samplers

Swiftb0y committed:
Apply suggestions from code review

Co-authored-by: Jan Holthuis <holthuis.jan@googlemail.com>

ronso0 committed:
track info on double click

Swiftb0y committed:
Changelog: add note on microphone ducking strength control

Co-authored-by: ronso0 <ronso0@mixxx.org>

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

uklotzde committed:
Merge pull request #3974 from Holzhaus/soundsourcemixin

test: Only register SoundSourceProviders when actually necessary

JosepMaJAZ committed:
tools/windows_buildenv.bar: Fix setting of buildenv variables

JosepMaJAZ committed:
tools/windows_buildenv.bat: Use 7z or powershell for unzipping

uklotzde committed:
Merge pull request #3969 from Holzhaus/pre-commit-githelper-move-fix

pre-commit: Fix detection of renamed files

ronso0 committed:
Waveforms on macOS: don't offer simple renderers (RGB, HSV, Filtered)

Swiftb0y committed:
CHANGELOG: add HE-AAC encoding feature

Holzhaus committed:
README: Fix name of x64 nativate tools prompt

Holzhaus committed:
tools/windows_buildenv: Improve command line output

Holzhaus committed:
tools/windows_buildenv.bat: Fix CMAKE_PREFIX_PATH

Holzhaus committed:
CMake: Read env vars from buildenv script

This fixes building Mixxx after running the windows buildenv script with
just `cmake -G Ninja ..` instead of requiring to pass various cmake
args.

Holzhaus committed:
tools/windows_buildenv.bat: Set Ninja generator from buildenv script

This makes the build instructions on all supported OSes
(Linux/macOS/Windows) identical. No special cases anymore.

Holzhaus committed:
CI: Remove generator from CMake command line

This is not necessary anymore. If it fails, our buildenv scripts are
insufficient, so this also helps testing these scripts.

Holzhaus committed:
CI: Remove unused QT_PATH env var from command line

Holzhaus committed:
README: Clarify build instructions

Holzhaus committed:
gitignore: Also ignore backup versions of CMakeSettings.json

ronso0 committed:
Skin context menus: decrease item padding a bit

Holzhaus committed:
CMake: Fix Qt5Keychain detection

ronso0 committed:
Skins: replace remaining px with em for the table header

ronso0 committed:
Skin qss: add/fix missing menu separators, fix font in LateNight

Holzhaus committed:
CMake: Fix setting of X_VCPKG_APPLOCAL_DEPS_INSTALL

ywwg committed:
Merge branch 'main' into stopped-explicit

daschuer committed:
Fix package dependency for ffmpeg

daschuer committed:
Fix not working cmake default_option() with braces in the default string

geovannimp committed:
[DDJ 200] Fix headphone master button led

The led button was turned on when disabled and off when enabled. This PR fix this behavior.

uklotzde committed:
Merge pull request #3979 from ronso0/macos-exclude-simple-renderers

Waveforms on macOS: don't offer simple renderers (RGB, HSV, Filtered)

daschuer committed:
Merge pull request #3968 from ronso0/library-scaling-finale

Skins: scale context menus and table header with system and library font

Holzhaus committed:
CI: Use custom CMake version on macOS/Ubuntu 18.04 only

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

# Conflicts:
#	src/waveform/widgets/waveformwidgettype.h

uklotzde committed:
Merge pull request #3981 from daschuer/fix_ppa_m4a

Fix ffmpeg dependency for our PPA builds

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

uklotzde committed:
Merge pull request #3983 from Holzhaus/ci-cmake-macos-only

CI: Use custom CMake version on macOS/Ubuntu 18.04 only

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

uklotzde committed:
CI: Remove references to ubuntu-18.04 from build

uklotzde committed:
CI: Upgrade jwlawson/actions-setup-cmake from v1.4 to v1.9

uklotzde committed:
Enable StatementMacros in .clang-format

uklotzde committed:
Delete obsolete comments referencing Ubuntu Bionic 18.04

uklotzde committed:
Delete obsolete workaround when casting a double into an enum

uklotzde committed:
Merge pull request #3929 from daschuer/qmlformat

only fail qmlformat stage if Qt >= 5.15 is used

Holzhaus committed:
skins/QMLDemo: Move image sources to theme singleton

ronso0 committed:
DlgPrefWaveforms: sort waveform types alphabetically

Holzhaus committed:
skins/QMLDemo: Add section background

ronso0 committed:
DlgPrefWaveforms: adjust ResetToDefault to new combobox content

Holzhaus committed:
skins/QMLDemo: Add sampler row

ronso0 committed:
LateNight: fix scaling of icons still using background-image

ronso0 committed:
Skins, waveforms: remove I-beam from in/outro marker to equalize marker size

Swiftb0y committed:
Merge pull request #3988 from ronso0/qss-icon-scaling

LateNight: fix scaling of icons still using background-image

uklotzde committed:
Merge pull request #3980 from ronso0/pref-sort-waveform-types

Waveforms preferences: sort waveform types combobox alphabetically

uklotzde committed:
Merge pull request #3982 from geovannimp/patch-4

[DDJ 200] Fix headphone master button led

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

# Conflicts:
#	src/waveform/widgets/waveformwidgettype.h

daschuer committed:
Revert extra if condition to add a default option and don't use not working and redundant braces.

daschuer committed:
Allow parentheses in default_option() depends string

Holzhaus committed:
QML: Add arcStart property to Knob control

Holzhaus committed:
skins/QMLDemo: Add basic effect units

Holzhaus committed:
skins/QMLDemo: Move effects to separate row

ronso0 committed:
Skins, waveforms: position in/outro markers inside the range

Holzhaus committed:
skins/QMLDemo: Add basic microphone/aux inputs to crossfader row

daschuer committed:
Merge pull request #3989 from ronso0/waveform-introoutro-markers

Skins, waveforms: remove I-beam from in/outro markers to equalize marker size

daschuer committed:
Merge pull request #3985 from uklotzde/obsolete-ubuntu-bionic-18.04

Obsolete Ubuntu Bionic 18.04

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

uklotzde committed:
SoundSourceFLAC: Fix armv7hl build on gcc 11.1

../src/sources/soundsourceflac.cpp: In member function 'FLAC__StreamDecoderWriteStatus mixxx::SoundSourceFLAC::flacWrite(const FLAC__Frame*, const FLAC__int32* const*)':
../src/sources/soundsourceflac.cpp:426:41: error: comparison of integer expressions of different signedness: 'mixxx::audio::SampleRate::value_t' {aka 'unsigned int'} and 'SINT' {aka 'int'} [-Werror=sign-compare]
  426 |     if (getSignalInfo().getSampleRate() != SINT(frame->header.sample_rate)) {
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors

Holzhaus committed:
skins/QMLDemo: Improve microphone/aux units and add ducking controls

Holzhaus committed:
skins/QMLDemo: Fix crossfader margins

JoergAtGithub committed:
CMake: Set default build type if not set explicitly

daschuer committed:
Merge pull request #3993 from uklotzde/armv7hl

SoundSourceFLAC: Fix armv7hl build on gcc 11.1

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

alexhorner committed:
Add AAC to dynamic metadata statement

Holzhaus committed:
skins/QMLDemo: Use explicit id in EffectUnit instead of "parent"

Holzhaus committed:
skins/QMLDemo: Move EmbeddedBackground color into component

daschuer committed:
Merge pull request #3899 from ywwg/stopped-explicit

Fix issues with half/double and Sync Lock

Holzhaus committed:
skins/QMLDemo: Move Microphone Ducking panel to separate component

Holzhaus committed:
QML: Introduce reusable PlayerDropArea component

Holzhaus committed:
skins/QMLDemo: Fix typo in MiniKnob foreground id

Holzhaus committed:
skins/QMLDemo: Remove unnecessary mousearea id from sampler

Holzhaus committed:
skins/QMLDemo: Introduce samplerColor

Holzhaus committed:
skins/QMLDemo: Use shared text components on samplers/decks/mic/aux

Holzhaus committed:
QML: Make Knob arc calculation more readable

uklotzde committed:
SoundSourceM4A: Fix armv7hl build on gcc 11.1

uklotzde committed:
SoundSourceM4A: Reduce log level for corrupt files

Errors are not critical unless Mixxx might crash. Errors in input
data should never be critical and must be handled.

alexhorner committed:
Precommit check fixes

Holzhaus committed:
skins/QMLDemo: Play on click, stop/eject on doubleclick

uklotzde committed:
DlgTrackInfo: Preserve existing track properties and metadata

uklotzde committed:
Preserve audio stream info when re-importing metadata

ronso0 committed:
Broadcasting: fix wrapping arund pushbutton values

ronso0 committed:
Skins: fix some Broadcasting styles

Swiftb0y committed:
Merge pull request #3992 from Holzhaus/qml-demo-skin-additions

QML demo skin additions

Swiftb0y committed:
Merge pull request #3997 from uklotzde/armv7hl

SoundSourceM4A: Fix armv7hl build on gcc 11.1

Swiftb0y committed:
QMLDemo: Add enum for specifing knob arc start position

ronso0 committed:
Pref > Deck: fix enable/disable pitch bend widgets

ronso0 committed:
Pref > Deck: consolidate pitch bend setup

ronso0 committed:
Pref > Deck: rename confusing variables in update()

ronso0 committed:
RateControl: remove redundant rampDirection condition

uklotzde committed:
SoundSourceMp3: Fix armv7hl build on gcc 11.1

Swiftb0y committed:
README: Add GH Actions Build badge for main branch

Holzhaus committed:
Merge pull request #4007 from Swiftb0y/readme-main-build-badge

README: Add GH Actions Build badge for main branch

Holzhaus committed:
CMake: Default to RelWithDebInfo builds on Windows

Holzhaus committed:
Merge pull request #4006 from uklotzde/armv7hl

SoundSourceMp3: Fix armv7hl build on gcc 11.1

Holzhaus committed:
Merge pull request #3998 from uklotzde/reimport-track-metadata

Fix reimport of track metadata

Holzhaus committed:
Merge pull request #3991 from daschuer/default_option_braces

cmake default_option() parentheses

Holzhaus committed:
Merge pull request #3976 from Swiftb0y/update-2.3-changelog

update 2.3 Changelog

Holzhaus committed:
Merge pull request #3966 from daschuer/buildenvhelp

help output for buildenv scripts

uklotzde committed:
Merge branch 'main' of git@github.com:mixxxdj/mixxx.git into source-synchronized-timestamp

# Conflicts:
#	src/track/trackrecord.cpp

ronso0 committed:
RateControl::processTempRate: use c++ style casts

Co-authored-by: Daniel Schürmann <daschuer@mixxx.org>

ronso0 committed:
Pref > Deck: improve rate ramp controls arrangement

Holzhaus committed:
CMake: Default to RelWithDebInfo

uklotzde committed:
Merge pull request #4001 from ronso0/broadcast-toggle

Broadcasting: fix wrapping arund pushbutton values

daschuer committed:
Merge pull request #3995 from alexhorner/main

Add AAC Dynamic Metadata

daschuer committed:
Added Alexander Horner to the contributor list in the about box. Thank you very much.

daschuer committed:
Merge remote-tracking branch 'upstream/2.3' into main

# Conflicts:
#	src/dialog/dlgabout.cpp

daschuer committed:
Merge pull request #3977 from Holzhaus/windows-buildenv-fix

Fix tools/windows_buildenv.bat and use 7z/PowerShell for extraction

daschuer committed:
Fail with CMAKE_BUILD_TYPE=MinSizeRel or any other unsupported value

daschuer committed:
Set /OPT:REF for "Release" builds as well

daschuer committed:
set STRINGS property for OPTIMIZE

daschuer committed:
Merge pull request #3978 from uklotzde/source-synchronized-timestamp

DB v37: Synchronize file modified time with track source on metadata import/export

daschuer committed:
Check if file is writable before making a copy for writing

daschuer committed:
Merge pull request #4002 from ronso0/rateramp-config

Preferences > Decks: fix activation of rate change controls

uklotzde committed:
Track: Warn when backdating source synchronization time stamps

uklotzde committed:
DB: Fix 0/NULL issues when reading source_synchronized_ms

uklotzde committed:
Track source sync: Ignore bogus file mopdification time stamps

uklotzde committed:
DB v38: Fix 0/NULL issue after upgrade to schema version 37

uklotzde committed:
Fix condition

...even though the edge is already handled before and will never occur.

ronso0 committed:
Merge pull request #4009 from daschuer/3977_follow_up

#3977 follow up - fix build env

uklotzde committed:
Merge pull request #4010 from daschuer/ceck_writable

Check if file is writable before making a copy for writing

uklotzde committed:
Add debug assertion to document assumptions about QVariant

uklotzde committed:
Fix undefined behavior of toMSecsSinceEpoch() if QDateTime is not valid

https://doc.qt.io/qt-5/qdatetime.html#toMSecsSinceEpoch

daschuer committed:
Merge pull request #4012 from uklotzde/source_synchronized_ms

DB: Fix 0/NULL issues when reading source_synchronized_ms

uklotzde committed:
Merge branch '2.3' of git@github.com:mixxxdj/mixxx.git

# Conflicts:
#	.github/workflows/build.yml
#	.gitignore
#	CMakeLists.txt

Holzhaus committed:
Merge pull request #4003 from Swiftb0y/qml-improve-knob-dx

QMLDemo: Add enum for specifing knob arc start position

daschuer committed:
Move Sandbox::checkSandboxed() back to MixxxMainWindow fixing a crash introduced by #3956

This is a band aid. The reintroduced issue of settings path access before checking the sandbox will be fixed in a follow up PR.

git-developer committed:
fix(BCR2000): fix backwards beatloop

uklotzde committed:
Merge pull request #4014 from git-developer/feature/controller-mapping-behringer-bcr2000

fix(BCR2000): fix backwards beatloop

daschuer committed:
Don't use coninical path on path that may not exists, fixing lp1921541

daschuer committed:
User can't select an non existing File or Dir in the open file dialog. Abort.

daschuer committed:
replace Yes/No with OK in the Sandbox dialogs

daschuer committed:
Describe pendig issues as comment

daschuer committed:
Move Sandbox::checkSandboxed() back to main.cpp

daschuer committed:
Improve comment

daschuer committed:
Improve comment

uklotzde committed:
Fix typos and reword comment

uklotzde committed:
Merge pull request #4013 from daschuer/sandbox_revert

Qickfix: Move Sandbox::checkSandboxed()

uklotzde committed:
Merge pull request #4015 from daschuer/lp1921541

Sandbox Dialog Fixes

yledoare committed:
Merge branch 'mixxxdj:main' into master

yledoare committed:
update mixxxVersion

yledoare committed:
update mixxxVersion

Holzhaus commented:
Why did you merge main? This makes it impossible to merge into 2.3. Please rebase on 2.3. and remove all commit from main.

yledoare commented:
> Why did you merge main? This makes it impossible to merge into 2.3. Please rebase on 2.3. and remove all commits from main.

This was only for testing.. sorry. Rebase like this :  "git rebase --onto upstream/2.3 upstream/pr/3246 Behringer-CMD-PL-1" ?

yledoare commented:
Maybe the simpler for me should be creating a new PR

Holzhaus commented:
> Maybe the simpler for me should be creating a new PR

No, this makes no difference because you still need to rebase, but makes review harder because discussion is split over multiple PRs.

uklotzde commented:
Any progress on this? Rescheduling to 2.3.2.

yledoare commented:
Hello,
I have only two files to add, for this controller working with Mixxx, please give me some instruction to add them by the simplest way

Regards.

git-developer commented:
Looks like this PR is obsolete for follow-up #4493 (targeting 2.3). Is this correct, @yledoare?

ronso0 commented:
superseded by #11518
